h2o-py/tests/testdir_hadoop/pyunit_INTERNAL_HADOOP_download_logs.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import print_function
+import sys
+sys.path.insert(1,"../../")
+import h2o
+from tests import pyunit_utils
+import os
+import zipfile
+
+log_level_suffixes = ["-1-trace.log",
+                      "-2-debug.log",
+                      "-3-info.log",
+                      "-4-warn.log",
+                      "-5-error.log",
+                      "-6-fatal.log",
+                      "-httpd.log"]
+
+log_level_suffixes_positions = {"TRACE" : 0,
+                                "DEBUG" : 1,
+                                "INFO"  : 2,
+                                "WARN"  : 3,
+                                "ERROR" : 4,
+                                "FATAL" : 5,
+                                "HTTPD" : 6}
+
+def log_files_prefix(node_log_zip_name):
+    node_ip = node_log_zip_name[:-4].split("_")[1]
+    node_port = node_log_zip_name[:-4].split("_")[2]
+    return "h2o_" + node_ip + "_" + node_port
+
+
+def log_files_expected_suffixes(log_level):
+    pos = log_level_suffixes_positions[log_level]
+    return log_level_suffixes[pos:]
+
+def expected_log_files(all_nodes_log_dir, all_nodes_log_zip_files, log_level):
+    expected_files = []
+    for node_log_name in all_nodes_log_zip_files:
+        log_file_dir = path_to_single_node_logs(all_nodes_log_dir, node_log_name)
+        for suffix in log_files_expected_suffixes(log_level):
+            expected_files.append(log_file_dir + "/" + log_files_prefix(node_log_name) + suffix)
+    return expected_files
+
+def path_to_single_node_logs(nodes_log_dir, node_log_name):
+    zip_ref = zipfile.ZipFile(nodes_log_dir + "/" + node_log_name, 'r')
+    node_log_dir = nodes_log_dir + "/" + node_log_name + "_logs"
+    zip_ref.extractall(node_log_dir)
+    zip_ref.close()
+    node_log_dir_extracted = node_log_dir
+    while not contains_log_file(node_log_dir_extracted):
+        node_log_dir_extracted = node_log_dir_extracted + "/" + os.listdir(node_log_dir_extracted)[0]
+    return node_log_dir_extracted
+
+def contains_log_file(dir):
+    for f in os.listdir(dir):
+        if os.path.isfile(dir + "/" + f) and f.endswith(".log"):
+            return True
+    return False
+
+def hadoop_download_logs():
+
+
+    # Check if we are running inside the H2O network by seeing if we can touch
+    # the namenode.
+    hadoop_namenode_is_accessible = pyunit_utils.hadoop_namenode_is_accessible()
+
+    if hadoop_namenode_is_accessible:
+        hdfs_name_node = pyunit_utils.hadoop_namenode()
+
+
+    zip_file = h2o.download_all_logs()
```

Comment:
Other tests usually specify the target directory directly

    results_dir = pyunit_utils.locate("results")

Reply:
I thought that this method is for finding the test data files, couldn't fit a good use for it in this test

Reply:
This is used in most places where the test stores something. It should be used here at least for consistency of the code base.

Just look for 'pyunit_utils.locate("results")' in the code base. Good example similar to your problem is pyunit_h2osave_mode.py.